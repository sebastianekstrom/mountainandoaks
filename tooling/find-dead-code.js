const { exec } = require("child_process");

const findDeadCode = () => {
  exec("ts-prune | wc -l", (error, stdout) => {
    const amount = parseInt(stdout.trim(), 10);

    if (error) {
      console.log(error);
      return process.exit(1);
    }

    if (amount !== 0) {
      const pluralized = amount === 1 ? "instance" : "instances";
      console.log(
        `ðŸš¨ ${amount} ${pluralized} of dead code found. Run 'ts-prune' for more details`
      );
      return process.exit(1);
    }

    console.log("âœ… No instances of dead code found");
  });
};

findDeadCode();
