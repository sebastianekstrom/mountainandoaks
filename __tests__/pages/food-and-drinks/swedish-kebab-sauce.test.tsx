import React from "react";
import { render, screen } from "../../../tooling/tests";
import SwedishKebabSauce from "pages/food-and-drinks/swedish-kebab-sauce";

describe("SwedishKebabSauce page", () => {
  it("renders title correctly", () => {
    render(<SwedishKebabSauce />);
    const title = screen.queryAllByText(/Swedish kebab sauce/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<SwedishKebabSauce />);
    const ingress = screen.queryAllByText(
      /I've finally managed to recreate the kebab sauce that's used/i
    );
    expect(ingress).toBeDefined();
  });

  it("renders the ingredients", () => {
    render(<SwedishKebabSauce />);
    const flourLabel = screen.queryAllByText(/Fanta/i);
    const waterLabel = screen.queryAllByText(/Mayonnaise/i);
    expect(flourLabel).toBeDefined();
    expect(waterLabel).toBeDefined();
  });

  it("renders the method steps", () => {
    render(<SwedishKebabSauce />);
    const step = screen.queryAllByText(
      /Add all the ingredients in a container/i
    );
    expect(step).toBeDefined();
  });
});
