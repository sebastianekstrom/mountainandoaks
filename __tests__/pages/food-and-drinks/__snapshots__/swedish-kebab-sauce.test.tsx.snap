// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwedishKebabSauce page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="flex flex-col"
  >
    <ul>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Sour cream
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          70
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Mayonnaise
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          50
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Fefferoni juice
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          19
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Fanta (orange flavor)
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          1.8
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Aromat
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          0.5
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Garlic powder
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          0.5
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Dried oregano
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          0.3
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Sambal Oelek (optional)
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          0.3
          g
        </p>
      </li>
    </ul>
  </div>
</section>
`;
